Tooltip: Don't render buttons tooltip when show button text labels is enabled

afercia opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->
Fixes https://github.com/WordPress/gutenberg/issues/55815

## What?
<!-- In a few words, what is the PR actually doing? -->
Buttons that already show visible text don't need a tooltiip that just repeats the visible text.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Unnecessarily clutters the UI.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
Use the `showTooltip` prop based on the `showIconLabels` preference.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->
- Edit a draft post.
- Use the Tab key to move focus to the Preview icon button.
- Observe the button still displays its tooltip.
- Go to Options > Preferences, and enable 'Show button text labels'.
- Move focus to the Preview button again.
- Observe the button does not display a tooltip.
- Publish the post.
- Repeat the test on the 'View Post' button.
- Go to the Site editor.
- Test the View button by repeating the steps above.

Note: the 'Save/Saved' button does have the same problem but for now we have to force its tooltip to always render [as explained in this comment](https://github.com/WordPress/gutenberg/issues/55815#issuecomment-1792327348).


### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->

afercia committed:
Don't render Preview button tooltip when show text labels is enabled.

afercia committed:
Don't render View button tooltip when show text labels is enabled.

afercia committed:
Fix comment block.

afercia commented:
Rebased on top of latest trunk. I'd appreciate a quick review @carolinan or @tellthemachines when you have a chance, please 🙏

tellthemachines reviewed (commented):
Thanks for fixing this @afercia ! It does keep popping up every time buttons are changed. 

I noticed there's also a tooltip showing for the "View Post" button that appears on published posts in the post editor:

<img width="323" alt="Screenshot 2023-11-22 at 1 54 05 pm" src="https://github.com/WordPress/gutenberg/assets/8096000/29e819f8-b545-4db8-a0d3-b0740c906b83">

Might be worth fixing that as part of this changeset too? I think it's in [this file](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-post/src/components/view-link/index.js).

afercia committed:
Don't render View Post button tooltip when show text labels is enabled.

afercia commented:
> I noticed there's also a tooltip showing for the "View Post" button

Oh yes thank you @tellthemachines, added it 👍 I updated the test instructions accordingly.

tellthemachines reviewed (approved):
Thanks for updating @afercia! LGTM ✅
